lib/plugins/aws/deployFunction/index.js

```diff
@@ -66,9 +66,15 @@ class AwsDeployFunction {
   }
 
   deployFunction() {
-    const artifactFileName = this.provider.naming
-      .getFunctionArtifactName(this.options.function);
-    const artifactFilePath = path.join(this.packagePath, artifactFileName);
+    const functionObject = this.serverless.service.getFunction(this.options.function);
+    const funcPackageConfig = functionObject.package || {};
+    let artifactFilePath = funcPackageConfig.artifact;
+    // if function artifact is not provided, derive the default artifact path
```

Comment:
I would suggest to switch this statement and assume that the user is not using an artifact at the function package level (since most of the time this would be the case).